src/runtime/io.cpp

```diff
@@ -291,6 +296,34 @@ obj_res decode_io_error(int errnum, b_obj_arg fname) {
     }
 }
 
+static obj_res mk_pipe_obj (int read, int write) {
+    object * res = lean_alloc_ctor(0, 2, 0);
+    lean_ctor_set(res, 0, lean_io_wrap_handle(fdopen(read, "r")));
+    lean_ctor_set(res, 1, lean_io_wrap_handle(fdopen(write, "w")));
+    return res;
+}
+
+/* Prim.mkPipe : IO Pipe := arbitrary _ */
+extern "C" obj_res lean_io_mk_pipe(obj_arg /* w */) {
+#if defined(LEAN_WINDOWS)
+    HANDLE read, write;
+    SECURITY_ATTRIBUTES sec_attr = { 0, nullptr, true };
+    bool success = CreatePipe(&read, &write, &sec_attr, 0);
+    if (!success) {
+        return set_io_error("Failed to create a pipe");
+    }
+    int in_fd = _open_osfhandle(reinterpret_cast<intptr_t>(read), _O_APPEND);
+    int out_fd = _open_osfhandle(reinterpret_cast<intptr_t>(read), _O_APPEND);
```

Comment:
This looks wrong. You are using `read` twice. Shouldn't one of them be `write` and the other `read`?

Reply:
Yes, you're right. I fixed it.